src/NServiceBus.AcceptanceTesting/Support/PublisherMetadata.cs

```diff
@@ -23,23 +23,19 @@ public void RegisterPublisherFor<T>(string endpointName)
         publisher.RegisterOwnedEvent<T>();
     }
 
-    public void RegisterPublisherFor<T>(Type endpointType)
-    {
-        RegisterPublisherFor<T>(Conventions.EndpointNamingConvention(endpointType));
-    }
+    public void RegisterSelfAsPublisherFor<TEventType>(object self) => RegisterPublisherFor<TEventType>(Conventions.EndpointNamingConvention(self.GetType()));
 
-    Dictionary<string, PublisherDetails> publisherDetails = [];
+    public void RegisterPublisherFor<TEventType>(Type endpointType) => RegisterPublisherFor<TEventType>(Conventions.EndpointNamingConvention(endpointType));
 
-    public class PublisherDetails
-    {
-        public PublisherDetails(string publisherName)
-        {
-            PublisherName = publisherName;
-        }
+    public PublisherDetails this[string publisherName] => publisherDetails.TryGetValue(publisherName, out var publisherDetail) ? publisherDetail : new PublisherDetails(publisherName);
+
+    readonly Dictionary<string, PublisherDetails> publisherDetails = [];
 
+    public class PublisherDetails(string publisherName)
+    {
         public List<Type> Events { get; } = [];
```

Comment:
Is it crazy to change this to a HashSet? It seems unlikely we break anyone.

Reply:
👍